databricks/koalas/CONTRIBUTING.md

```diff
@@ -0,0 +1,112 @@
+
+The `pandas.Series` object is much more versatile and universal than the `PySpark.Row` object. In particular, it can be used for most 
+practical purposes as a replacement, so it is the preferred way of returning single results, when they are not scalars.
+
+### Pandas functions that are not considered for inclusion in Koalas
+
+A few categories of functions are not considered for now to be part of the API, for different reasons:
+
+*Low level and deprecated functions* (Frame, user types, DataFrame.ix)
+
+*Functions that have an unexpected performance impact* 
+These functions (and the caller of theses functions) assume that the data is represented in a compact format (numpy in the case of pandas).
+Because these functions would force the full collection of the data and because there is a well-documented workaround, it is recommended that 
+they are not included. 
+
+The workaround is to force the materialization of the pandas DataFrame, either by calling:
+ - `.to_pandas()` (koalas only)
+ - `.to_numpy()` (works with both pandas and koalas)
+
+Here is a list of such functions:
+ - DataFrame.values
+ - `DataFrame.__iter__` and the array protocol `__array__`
+
+Other frameworks like Dask or Molin have a low-level block representation of a multidimensional array that Spark lacks. Until such representation is available, these functions should not be considered.
+
+### Spark functions that are temporarily included in Koalas
+
+- Column.alias
+- DataFrame.collect? `.to_pandas` is more complete
+- DataFrame.describe
+- DataFrame.distinct alias for `unique`
+- DataFrame.join alias for `.merge`
+- DataFrame.limit alias for `.head`
+
+### Spark functions that should be included in Koalas
+
+- pyspark.range
+
+Streaming functions
+
+Functions to control the partitioning, and in-memory representation of the data.
+
+Functions that add SQL-like functionalities to DataFrame.
+
+Functions that present algorithms specific to distributed datasets
+(approx quantiles for example)
+
+- DataFrame.approxQuantile, cache, checkpoint, coalesce, colRegex, createGlobalTempView, createOrReplaceGlobalTempView, createOrReplaceTempView, createTempView, crossJoin, crosstab, dropDuplicates / drop_duplicates, explain, hint, intersect, intersectAll, isLocal, isStreaming, localCheckpoint, persist, printSchema, registerTempTable, repartition, repartitionByRange, rollup, schema, select, selectExpr, subtract, take, toDF, unionByName, withWatermark
+
+
+### Reading and writing files.
+
+TODO: Koalas methods for reading and writing should work for both local and distributed files.
+
+
+When writing, the protocol to use is the default Spark protocol. This means in particular that
+dataframes written with Koalas will have multiple partitions, like in Spark. The current workaround is 
+to call `.to_pandas()` and then save it.
+
+## How to contribute
+
+The largest amount of work consists simply in implementing the pandas API in Spark terms, which is usually straightforward.
```

Comment:
I suggest that Koalas methods have the same parameters as Pandas methods, and we just raise a `NotImplementedError` for unsupported ones.